testscripts/nw_live_migration.ps1

```diff
@@ -0,0 +1,320 @@
+
+
+# Wait for migrate disk finished
+# Wait-Task -Task task$
+
+$vmObj = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+if (-not $vmObj) {
+    Write-Host -F Red "ERROR: Unable to Get-VM with $vmName"
+    Write-Output "ERROR: Unable to Get-VM with $vmName"
+    DisconnectWithVIServer
+    return $Aborted
+}
+
+
+# Another VM Ping During Migration
+Move-VM -VMotionPriority High -VM $vmObj -Destination (Get-VMHost $dstHost) -Confirm:$false -RunAsync:$true
+
+if (-not $?) {
+    Write-Host -F Red "ERROR : Cannot move VM to required Host $dstHost"
+    Write-Output "ERROR : Cannot move VM to required Host $dstHost"
+    DisconnectWithVIServer
+    return $Aborted
+}
+
+$command = "ping $ipv4 -c 50  | grep -i 'packet loss' | awk '{print `$(NF-4)}'"
+$packetLoss = Write-Output y | bin\plink.exe -i ssh\${sshKey} root@${ipv4Addr_B} $Command
+Write-Host -F Red "Info: Packets Loss $packetLoss"
+Write-Output "Info: Packets Loss $packetLoss"
+# DisconnectWithVIServer
+# return $Aborted
+
+
+# Wait for finish migrate if ping finshed first
+# Wait-Task -Task $task -ErrorAction SilentlyContinue
+
+$vmObj = Get-VMHost -Name $dstHost | Get-VM -Name $vmName
+if (-not $vmObj) {
+    Write-Host -F Red "ERROR: Unable to Get-VM with $vmName"
+    Write-Output "ERROR: Unable to Get-VM with $vmName"
+    DisconnectWithVIServer
+    return $Failed
+}
+
+
+Start-Sleep 1
+
+if ($packetLoss -ne "0%") {
+    Write-Host -F Red "ERROR : Packet Loss During Migration"
+    Write-Output "ERROR : Packet Loss During Migration"
+    $vmObj = Get-VMHost -Name $dstHost | Get-VM -Name $vmName
+    $task = Move-VM -VMotionPriority High -VM $vmObj -Destination (Get-VMHost $hvServer) -Confirm:$false
+    $vmObj = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+    # Move Hard Disk back to old datastore
+    $task = Move-VM -VMotionPriority High -VM $vmObj -Datastore $oldDatastore -Confirm:$false
+
+    DisconnectWithVIServer
+    return $Failed
+}
+
+# Clean up step
+
+# Move host to old host
+$task = Move-VM -VMotionPriority High -VM $vmObj -Destination (Get-VMHost $hvServer) -Confirm:$false
+
+if (-not $?) {
+    Write-Host -F Red "ERROR : Cannot move VM to required Host $dstHost"
+    Write-Output "ERROR : Cannot move VM to required Host $dstHost"
+    DisconnectWithVIServer
+    return $Failed
+}
+
+Start-Sleep 1
+
+$vmObj = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+
+if (-not $vmObj) {
+    Write-Host -F Red "ERROR: Unable to Get-VM with $vmName"
+    Write-Output "ERROR: Unable to Get-VM with $vmName"
+    DisconnectWithVIServer
+    return $Failed
+}
+
+
+# Move Hard Disk back to old datastore
+$task = Move-VM -VMotionPriority High -VM $vmObj -Datastore $oldDatastore -Confirm:$false
+
+if (-not $?) {
+    Write-Host -F Red "ERROR : Cannot move disk to required Datastore $oldDatastore"
+    Write-Output "ERROR : Cannot move disk to required Datastore $oldDatastore"
+    DisconnectWithVIServer
+    return $Failed
+}
+
```

Comment:
1. Move-VM twice and check packets loss twice
2. Packets loss value ~ 1%
3. Put $retVal = $PASSED into if-else

Reply:
Finished